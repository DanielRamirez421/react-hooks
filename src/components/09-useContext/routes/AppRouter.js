import React from 'react';
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from "react-router-dom";

import { AboutScreen } from '../screens/AboutScreen';
import { HomeScreen } from '../screens/HomeScreen';
import { LoginScreen } from '../screens/LoginScreen';
import { Navbar } from '../components/Navbar';


export const AppRouter = () => {
  return (
  <Router>
    <div>
      <Navbar />
      <hr />

      <div className='container'>
        <Routes>
          <Route exact path="/" element={ <HomeScreen /> }/>
          <Route exact path="/about" element={ <AboutScreen /> }/>
          <Route exact path="/login" element={ <LoginScreen /> }/>
          <Route path="*" element={ <HomeScreen /> }/>
        </Routes>
      </div>

    </div>
  </Router>
  );
};
