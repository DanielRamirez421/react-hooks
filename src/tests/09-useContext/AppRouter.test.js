import { mount } from "enzyme";
import { UserContext } from "../../components/09-useContext/context/UserContext";
import { AppRouter } from "../../components/09-useContext/routes/AppRouter";

describe('AppRouter component tests', () => {

  const user = { id: 123, name: 'Daniel', username: 'damtrax' };
  const wrapper = mount( 
    <UserContext.Provider value={{ user }}>
      <AppRouter /> 
    </UserContext.Provider>
  );

  test('should show AppRouter component', () => {
    expect(wrapper).toMatchSnapshot();
  });


});