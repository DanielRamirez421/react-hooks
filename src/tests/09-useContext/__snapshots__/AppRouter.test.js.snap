// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter component tests should show AppRouter component 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            activeclassname="is-active"
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            activeclassname="is-active"
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            activeclassname="is-active"
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <hr />
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 123,
   "name": "Daniel",
   "username": "damtrax"
}
      </pre>
    </div>
  </div>
</div>
`;
